src/hdmf/container.py

```diff
@@ -747,22 +747,34 @@ def __smart_str_dict(d, num_indent):
         out += '\n' + indent + right_br
         return out
 
-    def set_data_io(self, dataset_name: str, data_io_class: Type[DataIO], **kwargs) -> None:
+    def set_data_io(self, dataset_name: str, data_io_class: Type[DataIO], data_io_kwargs: dict = None, **kwargs):
         """
+        Apply DataIO object to a dataset field of the Container.
 
         Parameters
         ----------
         dataset_name: str
             Name of dataset to wrap in DataIO
         data_io_class: Type[DataIO]
             Class to use for DataIO, e.g. H5DataIO or ZarrDataIO
+        data_io_kwargs: dict, optional
```

Comment:
```suggestion
        data_io_kwargs: dict
```

@rly this should help a bit